entgql/template.go

```diff
@@ -476,7 +476,7 @@ func orderFields(n *gen.Type) ([]*OrderTerm, error) {
 			terms = append(terms, &OrderTerm{
 				GQL:   ant.OrderField,
 				Edge:  e,
-				Type:  e.Type,
+				Type:  n,
```

Comment:
`Type` is the ent type that owns the field. The previous version is correct. The problem is in the template that renders the code

Reply:
The owner of this order term passed/attached here, and I assume this will solve the issue https://github.com/ent/contrib/blob/master/entgql/template.go#L397

Reply:
I think to keep this Type value the same we'll have to add another field to this OrderTerm struct so that the template knows the type being sorted. Do you agree? I thought about that as a solution but I ended up thinking that the Type of the OrderType is the type of ent object being sorted, not the type of the edge. But perhaps that's not the way to think about it.

In the example of sorting a Todo by the Owner's name the old code would generate something like: 
```
GQL: OWNER_NAME,
Edge: User Edge
Type: User
Field: name
```
after the current change it'd be:
```
GQL: OWNER_NAME,
Edge: User Edge
Type: Todo
Field: name
```

In the case of sorting on a field, no edge, it'd be 
```
GQL: Text,
Edge: nil
Type: Todo
Field: text
```

If we keep this as is and instead add a new field to the OrderTerm struct, do you have any preference for what to call it? I don't think OwnerType is great because it's not necessarily the owner of the edge relationship. If we use the edge owner there is still a chance for conflict, we have to use the type of the object being sorted.

SortedType? EntType? Something else?